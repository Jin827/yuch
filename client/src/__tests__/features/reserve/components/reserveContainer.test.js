import React from 'react';
import { shallow } from 'enzyme';
import Buttons from '../../../../shared/buttons';
import ReserveContainer, {
  Unwrapped as UnwrappedReserveContainer,
} from '../../../../features/reserve/components/reserveContainer';
import ReserveForm from '../../../../features/reserve/components/reserveForm';

describe('<ReserveContainer />', () => {
  const setup = () => {
    const props = {
      show: false,
      variantValue: 'contained',
      colorValue: 'secondary',
      classNameVlue: 'bigButton',
      name: '예약하기',
    };

    const wrapper = shallow(<UnwrappedReserveContainer {...props} />);

    return { props, wrapper };
  };

  const { wrapper } = setup();

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('initializes reserve related state', () => {
    // expect(wrapper.state()).toEqual({data.initialState})
    // this test was done when action testing
  });

  describe('when clicking reserve button', () => {
    beforeEach(() => {
      wrapper.find(Buttons).simulate('click');
    });
    afterEach(() => {
      // wrapper.setState({show: false})
    });
    it('calls handleOpen callback', () => {});
    it('changes show state on button click event', () => {
      // expect.(wrapper.state().show).toEqual(true)
    });
    it('creates a reserve Modal component', () => {
      // expect(wrapper.find('SimpleModal').exists()).toBe(true);
    });
  });
});
