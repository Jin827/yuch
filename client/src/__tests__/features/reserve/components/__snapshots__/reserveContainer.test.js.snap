// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReserveContainer /> renders correctly 1`] = `
<div
  id="reserve"
>
  <div
    className="tc white reserve-container"
  >
    <h3
      className="white f-en"
    >
      Reservation
    </h3>
    <p>
      기업체 각종 행사, 모임 단체 식사 주문받습니다.
    </p>
    <WithStyles(Buttons)
      classNameValue="bigButton"
      colorValue="secondary"
      handleClick={[Function]}
      name="예약하기"
      variantValue="contained"
    />
  </div>
</div>
`;
